Build files are missing in published versions

**Description**

- Node.js version: 20.9.0
- Gitbeaker version: 39.33.2
- Gitbeaker release (cli, rest, core, requester-utils): all
- OS & version: Ubuntu 23.10

<!-- *(Brief description of your issue here)*

the new released versions have no source code in the packaged code
Error: Cannot find module '/home/alexander/.npm/_npx/fb4579aaf4722819/node_modules/@gitbeaker/core/dist/index.js'

(Paste any relevant logs - please use code blocks (```) to format console output,
logs, and code as it's very hard to read otherwise.) -->

**Steps to reproduce**

install newest version and use it

**Possible fixes**

add missing builded source code to the published packages

**Checklist**

- [x] I have checked that this is not a duplicate issue.
- [x] I have read the documentation.

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -454,7 +454,7 @@ release:special:
         labels=$(yarn auto label --pr $pr_id)
 
         if [[ "$labels" =~ "release:canary" ]]; then
-          yarn release:canary --dry-run
+          yarn release:canary
         fi
       fi
 

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -425,7 +425,6 @@ test:live:teardown:
     - ls -A packages/rest/dist
     - ls -A packages/cli/dist
 
-
 ## Canary or RC Special Releases
 
 release:special:

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -433,6 +433,9 @@ test:live:teardown:
     - git branch $CI_COMMIT_BRANCH HEAD
     - git checkout $CI_COMMIT_BRANCH
     - echo 'Ensure build files are present'
+
+    - rm -r packages/requester-utils/dist # Testing to make sure this fails
+
     - ls -A packages/requester-utils/dist
     - ls -A packages/core/dist
     - ls -A packages/rest/dist

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -196,15 +196,15 @@ test:live:setup:
     - echo "GITLAB_PERSONAL_ACCESS_TOKEN=${GITLAB_PERSONAL_ACCESS_TOKEN}" >> intergration.env
     - echo "Waiting for service to start"
     - sleep 180
+    - attempt=1
     - |
-      attempt=1;
-      while [[ "$(curl --fail --silent -X GET "$GITLAB_URL/-/readiness?all=1" --insecure | jq -r '.master_check[0].status')" != "ok" ]];
-      do
-        echo -ne "Polling Attempt: $attempt - Gitlab service still not alive yet!";
+      while [[ "$(curl --fail --silent -X GET "$GITLAB_URL/-/readiness?all=1" --insecure | jq -r '.master_check[0].status')" != "ok" ]]; do
+        echo -ne "Polling Attempt: $attempt - Gitlab service is not alive yet\033[0K\r";
         sleep 10;
         ((attempt++));
-      done;
-    - echo "Service is up and running!"
+      done
+    - printf "\n"
+    - echo "Service is up and running"
   artifacts:
     reports:
       dotenv: intergration.env

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -429,20 +429,20 @@ release:special:
     - test:unit:core
     - test:unit:cli
     - test:unit:rest
-    - test:types:utils
-    - test:types:core
-    - test:types:rest
-    - test:integration:rest
-    - test:integration:core
-    - test:e2e:cli
-    - test:e2e:rest
+    # - test:types:utils
+    # - test:types:core
+    # - test:types:rest
+    # - test:integration:rest
+    # - test:integration:core
+    # - test:e2e:cli
+    # - test:e2e:rest
   only:
     refs:
       - /^pr-[0-9]+$/
 
   script:
-    - echo 'release'
     - echo "Checking for special release..."
+    - ls -al packages/core/dist
     - |
       if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
         pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
@@ -453,7 +453,7 @@ release:special:
         labels=$(yarn auto label --pr $pr_id)
 
         if [[ "$labels" =~ "release:canary" ]]; then
-          yarn release:canary
+          yarn release:canary --dryRun
         fi
       fi
 

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -443,21 +443,20 @@ release:special:
 
   script:
     - echo "Checking for special release..."
-    - ls -al packages/core
     - ls -al ./packages/core/dist
-    # - |
-    #   if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
-    #     pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
+    - |
+      if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
+        pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
 
-    #     export CI_MERGE_REQUEST_ID=$pr_id
-    #     export CI_MERGE_REQUEST_SOURCE_BRANCH_NAME=$CI_COMMIT_BRANCH
+        export CI_MERGE_REQUEST_ID=$pr_id
+        export CI_MERGE_REQUEST_SOURCE_BRANCH_NAME=$CI_COMMIT_BRANCH
 
-    #     labels=$(yarn auto label --pr $pr_id)
+        labels=$(yarn auto label --pr $pr_id)
 
-    #     if [[ "$labels" =~ "release:canary" ]]; then
-    #       yarn release:canary --dryRun
-    #     fi
-    #   fi
+        if [[ "$labels" =~ "release:canary" ]]; then
+          yarn release:canary --dryRun
+        fi
+      fi
 
 ## Production
 

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -406,6 +406,19 @@ test:live:teardown:
     GIT_STRATEGY: clone
     GIT_DEPTH: 0
     NX_REJECT_UNKNOWN_LOCAL_CACHE: 0
+  needs:
+    - build # CHECKME: Needed for the artifacts
+    - test:unit:utils
+    - test:unit:core
+    - test:unit:cli
+    - test:unit:rest
+    - test:types:utils
+    - test:types:core
+    - test:types:rest
+    - test:integration:rest
+    - test:integration:core
+    - test:e2e:cli
+    - test:e2e:rest
   cache:
     - <<: *cacheinstallmodules
       policy: pull
@@ -429,59 +442,31 @@ test:live:teardown:
 
 release:special:
   extends: .release:base
-  needs:
-    - build # CHECKME: Needed for the artifacts
-    - test:unit:utils
-    - test:unit:core
-    - test:unit:cli
-    - test:unit:rest
-    - test:types:utils
-    - test:types:core
-    - test:types:rest
-    - test:integration:rest
-    - test:integration:core
-    - test:e2e:cli
-    - test:e2e:rest
   only:
     refs:
       - /^pr-[0-9]+$/
-
   script:
     - echo "Checking for special release..."
-    # - |
-    #   if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
-    #     pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
+    - |
+      if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
+        pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
 
-    #     export CI_MERGE_REQUEST_ID=$pr_id
-    #     export CI_MERGE_REQUEST_SOURCE_BRANCH_NAME=$CI_COMMIT_BRANCH
+        export CI_MERGE_REQUEST_ID=$pr_id
+        export CI_MERGE_REQUEST_SOURCE_BRANCH_NAME=$CI_COMMIT_BRANCH
 
-    #     labels=$(yarn auto label --pr $pr_id)
+        labels=$(yarn auto label --pr $pr_id)
 
-    #     if [[ "$labels" =~ "release:canary" ]]; then
-    #       yarn release:canary
-    #     fi
-    #   fi
+        if [[ "$labels" =~ "release:canary" ]]; then
+          yarn release:canary
+        fi
+      fi
 
 ## Production
 
 release:
   extends: .release:base
-  needs:
-    - build # CHECKME: Needed for the artifacts
-    - test:unit:utils
-    - test:unit:core
-    - test:unit:cli
-    - test:unit:rest
-    - test:types:utils
-    - test:types:core
-    - test:types:rest
-    - test:integration:rest
-    - test:integration:core
-    - test:e2e:cli
-    - test:e2e:rest
   only:
     refs:
       - main
       - next
-
   script: yarn release

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -419,50 +419,56 @@ test:live:teardown:
     - git remote set-url origin https://jdalrymple:${GITHUB_TOKEN}@github.com/jdalrymple/gitbeaker.git
     - git branch $CI_COMMIT_BRANCH HEAD
     - git checkout $CI_COMMIT_BRANCH
+    - echo 'Ensure build files are present'
+    - ls -A packages/requester-utils/dist
+    - ls -A packages/core/dist
+    - ls -A packages/rest/dist
+    - ls -A packages/cli/dist
+
 
 ## Canary or RC Special Releases
 
 release:special:
   extends: .release:base
   needs:
-    # - build
+    - build # CHECKME: Needed for the artifacts
     - test:unit:utils
     - test:unit:core
     - test:unit:cli
     - test:unit:rest
-    # - test:types:utils
-    # - test:types:core
-    # - test:types:rest
-    # - test:integration:rest
-    # - test:integration:core
-    # - test:e2e:cli
-    # - test:e2e:rest
+    - test:types:utils
+    - test:types:core
+    - test:types:rest
+    - test:integration:rest
+    - test:integration:core
+    - test:e2e:cli
+    - test:e2e:rest
   only:
     refs:
       - /^pr-[0-9]+$/
 
   script:
     - echo "Checking for special release..."
-    - ls -al ./packages/core/dist
-    - |
-      if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
-        pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
+    # - |
+    #   if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
+    #     pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
 
-        export CI_MERGE_REQUEST_ID=$pr_id
-        export CI_MERGE_REQUEST_SOURCE_BRANCH_NAME=$CI_COMMIT_BRANCH
+    #     export CI_MERGE_REQUEST_ID=$pr_id
+    #     export CI_MERGE_REQUEST_SOURCE_BRANCH_NAME=$CI_COMMIT_BRANCH
 
-        labels=$(yarn auto label --pr $pr_id)
+    #     labels=$(yarn auto label --pr $pr_id)
 
-        if [[ "$labels" =~ "release:canary" ]]; then
-          yarn release:canary
-        fi
-      fi
+    #     if [[ "$labels" =~ "release:canary" ]]; then
+    #       yarn release:canary
+    #     fi
+    #   fi
 
 ## Production
 
 release:
   extends: .release:base
   needs:
+    - build # CHECKME: Needed for the artifacts
     - test:unit:utils
     - test:unit:core
     - test:unit:cli

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -454,7 +454,7 @@ release:special:
         labels=$(yarn auto label --pr $pr_id)
 
         if [[ "$labels" =~ "release:canary" ]]; then
-          yarn release:canary --dryRun
+          yarn release:canary --dry-run
         fi
       fi
 

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -425,6 +425,7 @@ test:live:teardown:
 release:special:
   extends: .release:base
   needs:
+    - build
     - test:unit:utils
     - test:unit:core
     - test:unit:cli
@@ -442,20 +443,21 @@ release:special:
 
   script:
     - echo "Checking for special release..."
-    - ls -al packages/core/dist
-    - |
-      if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
-        pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
+    - la -al packages/core
+    - ls -al ./packages/core/dist
+    # - |
+    #   if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then
+    #     pr_id="${CI_COMMIT_BRANCH//[!0-9]/}"
 
-        export CI_MERGE_REQUEST_ID=$pr_id
-        export CI_MERGE_REQUEST_SOURCE_BRANCH_NAME=$CI_COMMIT_BRANCH
+    #     export CI_MERGE_REQUEST_ID=$pr_id
+    #     export CI_MERGE_REQUEST_SOURCE_BRANCH_NAME=$CI_COMMIT_BRANCH
 
-        labels=$(yarn auto label --pr $pr_id)
+    #     labels=$(yarn auto label --pr $pr_id)
 
-        if [[ "$labels" =~ "release:canary" ]]; then
-          yarn release:canary --dryRun
-        fi
-      fi
+    #     if [[ "$labels" =~ "release:canary" ]]; then
+    #       yarn release:canary --dryRun
+    #     fi
+    #   fi
 
 ## Production
 

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -425,7 +425,7 @@ test:live:teardown:
 release:special:
   extends: .release:base
   needs:
-    - build
+    # - build
     - test:unit:utils
     - test:unit:core
     - test:unit:cli

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -195,15 +195,14 @@ test:live:setup:
     - echo "GITLAB_URL=${GITLAB_URL}" >> intergration.env
     - echo "GITLAB_PERSONAL_ACCESS_TOKEN=${GITLAB_PERSONAL_ACCESS_TOKEN}" >> intergration.env
     - echo "Waiting for service to start"
-    - sleep 180
+    - sleep 210
     - attempt=1
     - |
       while [[ "$(curl --fail --silent -X GET "$GITLAB_URL/-/readiness?all=1" --insecure | jq -r '.master_check[0].status')" != "ok" ]]; do
-        echo -ne "Polling Attempt: $attempt - Gitlab service is not alive yet\033[0K\r";
+        echo "Polling Attempt: $attempt - Gitlab service is not alive yet";
         sleep 10;
         ((attempt++));
       done
-    - printf "\n"
     - echo "Service is up and running"
   artifacts:
     reports:

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -443,7 +443,7 @@ release:special:
 
   script:
     - echo "Checking for special release..."
-    - la -al packages/core
+    - ls -al packages/core
     - ls -al ./packages/core/dist
     # - |
     #   if [[ $CI_COMMIT_BRANCH =~ ^pr-[0-9]*$ ]]; then

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -432,9 +432,6 @@ test:live:teardown:
     - git branch $CI_COMMIT_BRANCH HEAD
     - git checkout $CI_COMMIT_BRANCH
     - echo 'Ensure build files are present'
-
-    - rm -r packages/requester-utils/dist # Testing to make sure this fails
-
     - ls -A packages/requester-utils/dist
     - ls -A packages/core/dist
     - ls -A packages/rest/dist

--- a/packages/cli/package.json
+++ b/packages/cli/package.json
@@ -38,8 +38,8 @@
     "lint:fix": "yarn lint --fix"
   },
   "dependencies": {
-    "@gitbeaker/core": "^39.33.2",
-    "@gitbeaker/rest": "^39.33.2",
+    "@gitbeaker/core": "39.33.2",
+    "@gitbeaker/rest": "39.33.2",
     "chalk": "4.1.2",
     "sywac": "^1.3.0",
     "xcase": "^2.0.1"

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -51,7 +51,7 @@
     "lint:fix": "yarn lint --fix"
   },
   "dependencies": {
-    "@gitbeaker/requester-utils": "^39.33.2",
+    "@gitbeaker/requester-utils": "39.33.2",
     "qs": "^6.11.2",
     "xcase": "^2.0.1"
   },

--- a/packages/rest/package.json
+++ b/packages/rest/package.json
@@ -53,8 +53,8 @@
     "lint:fix": "yarn lint --fix"
   },
   "dependencies": {
-    "@gitbeaker/core": "^39.33.2",
-    "@gitbeaker/requester-utils": "^39.33.2"
+    "@gitbeaker/core": "39.33.2",
+    "@gitbeaker/requester-utils": "39.33.2"
   },
   "devDependencies": {
     "@playwright/test": "^1.40.1",

